### PR TITLE
feat(install scripts): add automatic shell alias installation

### PR DESCRIPTION
**Pull request by taobojlen**
implement #90

## Description

This PR adds post-install and pre-uninstall scripts to prompt the user to automatically add shell aliases for npq upon installation, and to automatically remove those aliases when uninstalling. Initially just supports bash and zsh.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Related Issue

https://github.com/lirantal/npq/issues/90

## Motivation and Context

> When someone installs npq they would probably want to use it in an integrated manner with npm and yarn by aliasing it. (https://github.com/lirantal/npq/issues/90)

## How Has This Been Tested?

I've added integration tests for the two scripts, and unit tests for the shell detection code.
I haven't tested an interactive script like this one before, so I'm very happy to hear feedback here -- also on the fact that I couldn't quite get to the 90% coverage. The lines that aren't covered are related to file read/write error handling, and it seems like adding test cases with e.g. files you don't have permission to read will complicate things more than is worth it. But let me know if you disagree!

This has been tested in bash and zsh on my Linux machine.

## Other

I removed Node 8 support to use `fs.promises`, but this isn't strictly necessary and I'm happy to revert to plain `fs` calls if you'd like to keep supporting Node 8.

## Screenshots (if appropriate):
Screenshot from 2020-10-12 14-25-57

## Checklist:

- [x] I have updated the documentation (if required).
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by lirantal** [comment]
thanks @taobojlen, this looks great.
I feel that upon installation that message about adding npq to the shell might seem very obsecure and concerning to the user. How about we add a line before this to describe why we're doing it? something like:

```
Thank you for installing npq! We want to help you make conscious decisions before installing potentially dangerous packages.
To do that, we can add an npq as an alias to npm, so that `npm install <package>` will first use npq to verify the package and prompt you if it finds any issues.

Looks like your using zsh...
```

WDYT?

**Comment by taobojlen** [comment]
Yes, good point! I was meaning to ask about the wording, but forgot. Looks better now:

Screenshot from 2020-10-13 10-45-58

**Comment by lirantal** [comment]
Looks great! love it ✨

File: __tests__/scripts.test.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+const fs = require('fs')
+const inquirer = require('inquirer')
+
+const postinstall = require('../scripts/postinstall').testable
+const preuninstall = require('../scripts/preuninstall').testable
+const helpers = require('../scripts/scriptHelpers')
+
+const TEST_PROFILE_PATH = '/tmp/.npq_test_profile'
```
Comment (lirantal) [inline __tests__/scripts.test.js:8]: this test will only run on *nix based systems. there's an `os` module which you can use to get a tmpdir and then use `path` module to get the path for OS-specific.

**Review by lirantal — APPROVED** [review]

File: __tests__/scripts.test.js (hunk 1 above)
Comment (lirantal) [reply __tests__/scripts.test.js:8]: @taobojlen I'll merge for now but if you wanted to send a new PR to fix up the tests that would be nice

**Comment by lirantal** [comment]
@taobojlen the CI job to install dependencies is actually trying to run the postinstall script :-)

https://github.com/lirantal/npq/blob/master/.travis.yml#L9

**Comment by lirantal** [comment]
Resolved with https://github.com/lirantal/npq/pull/153

**Pull request merged**